### PR TITLE
Adjust audit policy to omit high-volume or large responses from expected actors

### PR DESCRIPTION
**Pull request by liggitt**
Builds on https://github.com/kubernetes/kubernetes/pull/52960

Possibly related to density test failures https://github.com/kubernetes/kubernetes/issues/51899#issuecomment-331924016

### CODE REVIEW

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -577,6 +577,31 @@ rules:
       - group: "" # core
         resources: ["events"]
 
+  # node and pod status calls from nodes are high-volume and can be large, don't log responses for expected updates from nodes
+  - level: Request
+    users: ["kubelet"] # legacy kubelet identity
```
Comment (shyamjvs) [inline cluster/gce/gci/configure-helper.sh]: I think NPD is doing this too quite frequently.

**Comment by crassirostris** [comment]
/cc @tallclair @destijl

Please review

**Comment by soltysh** [comment]
The policy change lgtm

**Comment by shyamjvs** [comment]
Policy lgtm - will let you know if there's sth else bothering at scale (none that I can see for now). It'd be good if we can get this merged asap.

**Comment by crassirostris** [comment]
OK, this seems fine anyway

/lgtm

**Comment by shyamjvs** [comment]
@liggitt Could you (maybe in a separate PR) move the policy out into a separate file so we can reuse it for kubemark?

**Comment by shyamjvs** [comment]
I'll send out a PR enabling it for kubemark.

**Comment by crassirostris** [comment]
@shyamjvs It will lose in readability when moved to a file. Maybe it can be extracted to the shared bash method?

**Comment by liggitt** [comment]
added alternate username for node-problem-detector for GCE deployment

**Comment by shyamjvs** [comment]
@crassirostris If we can reasonably do that, I'm fine with it (hopefully not needing to do extra bash sourcing from kubemark scripts).\
Ideally it would be cleaner to have it in a file - since it seems to be a static config anyway.

**Comment by liggitt** [comment]
> @shyamjvs It will lose in readability when moved to a file. Maybe it can be extracted to the shared bash method?

In general, when we have code that generates policies, I'd like to see checked-in versions of those (kept up to date via a test that compares the generated version with the committed version)

This lets people pull the default policy at a given point in time, and audit changes to that policy over time.

for example:
* Policy: https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go#L141
* rendered version: https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/testdata/cluster-roles.yaml
* test to ensure the rendered version stays up to date: https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy_test.go#L276

**Comment by shyamjvs** [comment]
@liggitt Any reason why not to just have a file with the policy (which is the one committed and also the one that'd be used)? Is it that we want to have more and more stuff generated?

File: cluster/gce/gci/configure-helper.sh [hunk 2]
```diff
@@ -577,6 +577,31 @@ rules:
       - group: "" # core
         resources: ["events"]
 
+  # node and pod status calls from nodes are high-volume and can be large, don't log responses for expected updates from nodes
+  - level: Request
+    users: ["kubelet", "system:node-problem-detector", "system:serviceaccount:kube-system:node-problem-detector"]
+    verbs: ["update","patch"]
+    resources:
+      - group: "" # core
+        resources: ["nodes", "nodes/status", "pods", "pods/status"]
```
Comment (tallclair) [inline cluster/gce/gci/configure-helper.sh]: Why are `nodes` and `pods` included here? I'm OK with it being request level, but I'm curious what update/patch calls to those objects are frequent.

File: cluster/gce/gci/configure-helper.sh [hunk 3]
```diff
@@ -577,6 +577,31 @@ rules:
       - group: "" # core
         resources: ["events"]
 
+  # node and pod status calls from nodes are high-volume and can be large, don't log responses for expected updates from nodes
+  - level: Request
```
Comment (liggitt) [inline cluster/gce/gci/configure-helper.sh]: if required to hit scale targets, we could even go to metadata-level logging on these, though I'd prefer to see the request content

**Comment by crassirostris** [comment]
@liggitt 

> In general, when we have code that generates policies, I'd like to see checked-in versions of those (kept up to date via a test that compares the generated version with the committed version)

That sounds perfect, but unfortunately the audit policy generation function is in the bash scripts

We can actually make it a default policy, not just GCE-specific. It then can be generated in the go code and a static version can be generated, similarly to RBAC policies. @soltysh @sttts WDYT about this?

**Comment by liggitt** [comment]
> We can actually make it a default policy, not just GCE-specific. It then can be generated in the go code and a static version can be generated, similarly to RBAC policies. @soltysh @sttts WDYT about this?

It references user names only set up by GCE. It is not appropriate as a generic policy.

**Comment by liggitt** [comment]
> unfortunately the audit policy generation function is in the bash scripts

we can move it :)

**Comment by crassirostris** [comment]
@liggitt 

> It references user names only set up by GCE. It is not appropriate as a generic policy.

OK, I see

**Comment by crassirostris** [comment]
> we can move it :)

Where?

File: cluster/gce/gci/configure-helper.sh (hunk 2 above)
Comment (liggitt) [reply cluster/gce/gci/configure-helper.sh]: rechecked output of e2e runs and you're right, only the status ones were high volume. update.

**Comment by liggitt** [comment]
https://github.com/kubernetes/kubernetes/issues/52560 flake
/retest

**Comment by liggitt** [comment]
/assign @mikedanese @roberthbailey
for gce approval

**Pull request merged**